resources/assets/coffee/react/beatmap-discussions/nominations.coffee

```diff
@@ -0,0 +1,126 @@
+
+  doAjax: (action, comment) ->
+    LoadingOverlay.show()
+
+    params =
+      method: 'PUT'
+
+    if comment
+      params['data'] = {'comment': comment}
+
+    $.ajax laroute.route("beatmapsets.#{action}", beatmapsets: @props.beatmapset.beatmapset_id), params
+
+    .done (response) =>
+      $.publish 'beatmapset:update', beatmapset: response.beatmapset.data
+
+    .fail osu.ajaxError
+    .always LoadingOverlay.hide
+
+  componentDidMount: ->
+    $.subscribe 'beatmapset:nominate.beatmapset-nominations', @nominate
+    $.subscribe 'beatmapset:disqualify.beatmapset-nominations', @disqualify
+
+  componentWillUnmount: ->
+    $.unsubscribe '.beatmapset-nominations'
+
+  render: ->
+    @reboot()
+
+    disqualificationBanner =
+      if @disqualification
+        div className: 'osu-layout__row osu-layout__row--sm1 osu-layout__row--page-compact',
+          div className: "#{bdn}__disqualification-banner",
+            span null,
+              "#{@disqualification.reason}"
+            span className: "#{bdn}__disqualification-time", dangerouslySetInnerHTML:
+              __html: osu.trans 'beatmaps.nominations.disqualifed-at', time_ago: osu.timeago(@disqualification.created_at)
+
+    nominationBanner =
+      if @mapCanBeNominated or @mapIsQualified
+        div className: bdn,
+          disqualificationBanner
+
+          div className: "osu-layout__row osu-layout__row--sm1 osu-layout__row--page-compact",
+            div className: "#{bdn}__message-area#{(if @mapIsQualified then '-qualified' else '')}",
+              span className: "#{bdn}__message-text",
+                if @mapCanBeNominated
+                  osu.trans 'beatmaps.nominations.required-text',
+                    current: @currentNominations,
+                    required: @nominationsRequired
+                else if @mapIsQualified
+                  if @rankingETA
+                    span dangerouslySetInnerHTML:
+                      __html: osu.trans 'beatmaps.nominations.qualified', date: osu.timeago(@rankingETA)
+                  else
+                    span dangerouslySetInnerHTML:
+                      __html: osu.trans 'beatmaps.nominations.qualified-soon'
+
+              if @userCanPerformNominations
+                span className: "#{bdn}__button-area",
+                  if @mapIsQualified
+                    button
+                      className: 'btn-osu-lite btn-osu-lite--pink'
+                      onClick: => $.publish 'beatmapset:disqualify', all: 'disqualify'
+                      el Icon, name: 'thumbs-down'
+                      span className: "#{bdn}__button-text",
+                        osu.trans 'beatmaps.nominations.disqualify'
+                  else if @mapCanBeNominated
+                    button
+                      className: 'btn-osu-lite btn-osu-lite--green'
+                      disabled: @alreadyNominated
+                      onClick: => $.publish 'beatmapset:nominate', all: 'nominate'
```

Comment:
apparently binding `this` (or fat arrow in this case) for props is a [bad thing™](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md).